modules/tensor_mechanics/src/materials/ComputeThermalExpansionEigenstrainBase.C

```diff
@@ -34,9 +34,10 @@ ComputeThermalExpansionEigenstrainBaseTempl<is_ad>::ComputeThermalExpansionEigen
     _use_old_temperature(this->template getParam<bool>("use_old_temperature")),
     _temperature_old(this->_fe_problem.isTransient() ? this->coupledValueOld("temperature")
                                                      : this->_zero),
-    _deigenstrain_dT(is_ad ? nullptr
-                           : &this->template declarePropertyDerivative<RankTwoTensor>(
-                                 _eigenstrain_name, this->getVar("temperature", 0)->name())),
+    _deigenstrain_dT((is_ad || !(this->getVar("temperature", 0)))
+                         ? nullptr
+                         : &this->template declarePropertyDerivative<RankTwoTensor>(
+                               _eigenstrain_name, this->getVar("temperature", 0)->name())),
```

Comment:
we should have a `isCoupledConstant` helper method in `Coupleable` to make such code more readable.

Reply:
Yeah, we had a discussion in the referenced issue. Why don't you join the party there?